### ISSUE TITLE
Literal types with generic wrapper functions

### ISSUE THREAD

**Issue by dbrattli** [microsoft/pyright#3086]
**Describe the bug**

Using literal types through generic wrapper functions produces type errors. I'm currently unsure if this is a bug or by design. I think it's a bug since I can use literals with normal generic functions. It would be great if you could have a look:

**To Reproduce**
Steps to reproduce the behavior.

In the code below there are 3 wrapper functions `wrapper1`, `wrapper2` and `wrapper3`. Currently only `wrapper1` will successfully type check when wrapping, but I would expect that using `wrapper2` or `wrapper3` should also type check. The error I get is:

```
Argument of type "(a: ColumnType, b: int, c: int) -> int" cannot be assigned to parameter "fn" of type "(_A@wrapper2, **_P@wrapper2) -> _B@wrapper2" in function "wrapper2"
  Type "(a: ColumnType, b: int, c: int) -> int" cannot be assigned to type "(_A@wrapper2, **_P@wrapper2) -> _B@wrapper2"
    Parameter 1: type "_A@wrapper2" cannot be assigned to type "ColumnType"
      "ColumnType" cannot be assigned to type "ColumnType"
```

**Screenshots or Code**

```python
from typing import Callable, TypeVar, Literal
from typing_extensions import (
    Concatenate,
    ParamSpec,
)

_P = ParamSpec("_P")
_A = TypeVar("_A")
_B = TypeVar("_B")
_C = TypeVar("_C")
_D = TypeVar("_D")

ColumnType = Literal["order_id"]

def wrapper1(fn: Callable[_P, _B]) -> Callable[_P, _B]:
    return fn

def wrapper2(
    fn: Callable[Concatenate[_A, _P], _B]
) -> Callable[Concatenate[_A, _P], _B]:
    return fn

def wrapper3(fn: Callable[[_A, _B, _C], _D]) -> Callable[[_A, _B, _C], _D]:
    return fn

@wrapper1
def f1(a: ColumnType, b: int, c: int) -> int:
    return c

@wrapper2
def f2(a: ColumnType, b: int, c: int) -> int:
    return c

@wrapper3
def f3(a: ColumnType, b: int, c: int) -> int:
    return c
```

**VS Code extension or command-line**

VS Code extension 1.1.222

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.223, which I just published. It will also be included in the next release of pylance.
